bundles/org.palladiosimulator.commons.ui/src/org/palladiosimulator/commons/ui/launch/ObservableLaunchConfigurationAdapter.java

```diff
@@ -0,0 +1,119 @@
+package org.palladiosimulator.commons.ui.launch;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.eclipse.core.databinding.observable.sideeffect.ISideEffect;
+import org.eclipse.core.databinding.observable.value.IObservableValue;
+import org.eclipse.core.databinding.observable.value.WritableValue;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+
+/**
+ * This class serves as generic bridge between Eclipse Launch Configurations and the Eclipse databinding framework.
+ * 
+ * For detaiuls on its usage, please see {@link ObservableLaunchConfigurationAttributeFactory}.
+ * 
+ * @author Sebastian Krach
+ *
+ */
+public class ObservableLaunchConfigurationAdapter implements ObservableLaunchConfigurationAttributeFactory {
+    protected Map<String, Consumer<ILaunchConfigurationWorkingCopy>> lcUpdates = new HashMap<>();
+    protected Map<String, Consumer<ILaunchConfiguration>> mUpdates = new HashMap<>();
+    protected Map<String, Consumer<ILaunchConfigurationWorkingCopy>> defaults = new HashMap<>();
+    protected Map<String, IObservableValue<?>> observables = new HashMap<>();
+    private ISideEffect dirtyNotificationEffect;
+    private Runnable dirtyNotification;
+
+    public void notifyWhenDirty(Runnable dirtyNotification) {
+        this.dirtyNotification = dirtyNotification;
+    }
+
+    public void initializeFrom(final ILaunchConfiguration configuration) {
+        if (dirtyNotificationEffect != null) {
+            dirtyNotificationEffect.dispose();
+        }
+        mUpdates.values()
+            .forEach(c -> c.accept(configuration));
+        dirtyNotificationEffect = ISideEffect.create(() -> {
+            observables.values()
+                .forEach(IObservableValue::getValue);
+            dirtyNotification.run();
+        });
+    }
+
+    public void performApply(ILaunchConfigurationWorkingCopy workingCopy) {
```

Comment:
After thinking about that for a few moments, the meaning of the method becomes clear. However, some JavaDoc would be appreciated.